### ISSUE TITLE
ui Path length doesn't align with standard formula

### ISSUE THREAD

**Issue by JKris95** [flutter/flutter#108130]
The length of a circle as reported by the `length` property of `dart:ui`'s `PathMetric` class deviates from the circle length calculated by the standard circle circumference formula `c = 2 * pi * r`

## Steps to Reproduce

1. Create a new Flutter dartpad
2. Copy the code sample below and paste it into the dartpad
3. Run the example.

**Expected results:**
I expected that the two lengths would (almost) equal each other.

**Actual results:**
As printed by the example code, there is a deviation between the two length calculations. From length calculations with 1000 different radii ranging from 1 to 1000, I find the mean deviation to be ~0.56 and the max deviation to be ~1.25.

I am not sure if this is a bug or not. If it is expected I am hoping that the reason can be explained and possibly added to the documentation. The documentation for path metric length currently reads:

> Return the total length of the current contour.

<details>
<summary>Code sample</summary>

```
import 'dart:ui';
import 'dart:math';

void main() {
  double maxCircleLengthDeviation = 0.0;
  double maxDeviationRadius = 1.0;
  double deviationSum = 0.0;
  const int iterations = 1000;
  for (int i = 1; i <= iterations; i++) {
    final double radius = i.toDouble();

    final Path circle = Path()
      ..addArc(Rect.fromCircle(center: const Offset(10, 10), radius: radius),
          0.0, 2 * pi);

    final double circlePathLength = getSingleContourPathLength(circle);
    final double standardCircleLength = pi * 2 * radius;

    final deviation = (circlePathLength - standardCircleLength).abs();
    deviationSum += deviation;
    
    if (deviation > maxCircleLengthDeviation) {
      maxCircleLengthDeviation = deviation;
      maxDeviationRadius = radius;
    }
  }
  
  print('max circle length deviation: $maxCircleLengthDeviation at radius $maxDeviationRadius');
  
  final double circleLengthMeanDeviation = deviationSum / iterations;
  print('mean deviation: $circleLengthMeanDeviation');
}

double getSingleContourPathLength(Path path) {
  final metrics = path.computeMetrics().toList();
  if (metrics.length != 1) {
    throw ArgumentError('path must have exactly one contour');
  }
  return metrics.fold(0.0,
      (double tmpLength, PathMetric contour) => tmpLength + contour.length);
}
```
</details>

I have run the example on all channels available for dartpad.  The output was consistent across all channels.

- Stable channel: Flutter 3.0.5 Dart SDK 2.17.6
- Beta channel: Flutter 3.1.0 Dart SDK 2.18.0-44.1.beta
- Old channel: Flutter 2.10.5 Dart SDK 2.16.2

**Comment by huycozy** [comment]
Hi @JKris95 
Regrading to your filed issue about the length of contour here #107965, I wonder if that issue causes your deviant calculations on this issue?

**Comment by JKris95** [comment]
Hi @huycozy.
Without knowing the details of the path implementation I see the two issues as unrelated. Where #107965 is in regard to the actual number of contours (contour count) in a path this issue is questioning the length of a contour, more specifically the circumference of a circle.

I have also checked the length of a line after adding one to the path with the `lineTo` method on `Path`.  Generally, line length was much closer to the result of a standard vector length calculation although there were significant deviations depending on s. I used the `distance` property of the `Offset` class from `dart:ui` for line length comparison
